docs/DeveloperGuide.adoc

```diff
@@ -385,6 +385,225 @@ We are using `java.util.logging` package for logging. The `LogsCenter` class is
 
 Certain properties of the application can be controlled (e.g App name, logging level) through the configuration file (default: `config.json`).
 
+=== Automatic creation of tags from person attributes
+
+==== Creation of the `Tag`
+We have made it such that the attributes `Price`, `Subject`, `Level` and `Status` of a person each have a tag with the value added to the person.
+This is done when user performs `add` or `edit` command.
+
+In the AddCommandParser, after successfully parsing the `Price`, `Subject`, `Level`, `Status` input, the code has been modified such that a new `Tag` containing the correct value and attribute is automatically created after.
+For the EditCommand, the `createEditedPerson` command will remove the previous attribute tags in the `Person` list of tags, then recreate the attribute tags using the new `Person` class attribute values.
+
+The idea for this is implemented this way:
+[source,java]
+----
+private static Person createEditedPerson(...) {
+	/* code to update the attributes here */
+    Set<Tag> attributeTags = new HashSet<>(updatedTags);
+	attributeTags.remove(new Tag(personToEdit.getPrice().toString()));
+	/* similar code to remove the other attribute tags... */
+	attributeTags.add(new Tag(updatedPrice.toString(), Tag.AllTagTypes.PRICE));
+	/* similar code to add the new attribute tags... */
+	return new Person(...)
+}
+----
+[NOTE]
+Since the `Price`, `Subject`, `Level` and `Status` attributes have been parsed and checked to conform to expected values, there is no need to reparse the values when creating them as a `Tag`. +
+The acceptable inputs for a `Tag` must necessarily allow all the acceptable inputs for `Price`, `Subject`, `Level` and `Status` to prevent errors.
+
+==== `Tag` class attributes
+The `Tag` class can support the ability to classify which attribute the tag was created from. This is done by adding an enum `AllTagsType` to the `Tag` class.
+A tag can be created without being from an attribute, with just a tag name. In this case, creating a `Tag` with no provided `AllTagsType` will initialise the `tagType` to `DEFAULT`.
+
+This is the current `Tag` class in UML notation.
+
+image::TagClassDiagram.png[width="600"]
+
+This new attribute is used to implement the capability to standardize all tags created as a `Price` attribute to have the same color, all tags created as a `Subject` attribute to have the same color, and so on.
+This has been implemented as a new overloaded `getTagColorStyleFor` method, which takes in a Tag instead of a String.
+
+The method idea is implemented this way:
+[source,java]
+----
+private String getTagColorStyleFor(Tag tag) {
```

Comment:
Please use code snippets minimally.
Reason: it is difficult to maintain these in the document, particularly if the implementation changes.